server/api/user/user.controller.js

```diff
@@ -295,17 +296,30 @@ exports.authCallback = function(req, res, next) {
 /**
  * Mark attendance for specified user
  */
-exports.attendance = function(req,res){
+exports.attend = function(req,res){
     var userId = req.params.id;
-    var result = User.update({
+    var code = req.body.dayCode;
+    if (!code) res.send(400, "No Code Submitted");
+    // Check code against current class year
+    ClassYear.findOne({
+      current: true
+    }, function(err, classYear){
+      if (err) return res.send(500, err);
+      if (!classYear) return res.send(500, "No Class Year");
+      if (classYear.dayCode === code){
```

Comment:
The class year represents the current semester of rcos, all day codes are recorded along with some other information.

A lot of this is confusing because the utility methods aren't in